app/src/main/java/com/poupa/vinylmusicplayer/dialogs/helper/DeleteSongsHelper.java

```diff
@@ -0,0 +1,223 @@
+package com.poupa.vinylmusicplayer.dialogs.helper;
+
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Build.VERSION_CODES;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.helper.MusicPlayerRemote;
+import com.poupa.vinylmusicplayer.misc.WeakContextAsyncTask;
+import com.poupa.vinylmusicplayer.model.Song;
+import com.poupa.vinylmusicplayer.ui.activities.saf.SAFGuideActivity;
+import com.poupa.vinylmusicplayer.util.MusicUtil;
+import com.poupa.vinylmusicplayer.util.PreferenceUtil;
+import com.poupa.vinylmusicplayer.util.SAFUtil;
+
+
+public class DeleteSongsHelper {
+
+    public static void delete(Song song, @NonNull FragmentManager manager, @Nullable String tag) {
+        if (Build.VERSION.SDK_INT <= VERSION_CODES.Q) { // API less or equal to 29
+            DeleteSongsDialogAndroidQ.create(song).show(manager, tag);
```

Comment:
if I follow this logic, the class should be named 'DeleteSongsDialogBeforeAndroidR', exact?

Reply:
Well there is an else to the function, so that this can manage all delete cases

Reply:
Well there is an else to the function, so that this can manage all delete cases